purchase_merge/wizard/purchase_merge.py

```diff
@@ -0,0 +1,226 @@
+            res["dst_purchase_id"] = self._get_ordered_purchase(active_ids)[-1].id
+        return res
+
+    # ----------------------------------------
+    # Check method
+    # ----------------------------------------
+
+    def _check_all_values(self, purchase_orders):
+        """Contain all check method"""
+        self._check_supplier(purchase_orders)
+        self._check_state(purchase_orders)
+        self._check_content(purchase_orders)
+
+    def _check_supplier(self, purchase_orders):
+        suppliers = purchase_orders.mapped("partner_id")
+        if len(suppliers) > 1:
+            raise UserError(
+                _("You can't merge PO's with different suppliers {}").format(
+                    ", ".join(suppliers.mapped("name"))
+                )
+            )
+
+    def _check_state(self, purchase_orders):
+        non_draft_po = purchase_orders.filtered(lambda p: p.state != "draft")
+        if non_draft_po:
+            po_names = non_draft_po.mapped("name")
+            raise UserError(
+                _(
+                    "You can't merge purchase orders that aren't in draft state like: {}"
+                ).format(po_names)
+            )
+
+    def _check_content(self, purchase_orders):
+        start_msg = _("You can't merge a purchase with different ")
+        error_msg = ""
+
+        currency_names = set(purchase_orders.mapped(lambda o: o.currency_id.name))
+        picking_type_names = set(
+            purchase_orders.mapped(lambda o: o.picking_type_id.name)
+        )
+        incoterm_names = set(purchase_orders.mapped(lambda o: o.incoterm_id.name))
+        payment_term_names = set(
+            purchase_orders.mapped(lambda o: o.payment_term_id.name)
+        )
+        fiscal_position_names = set(
+            purchase_orders.mapped(lambda o: o.fiscal_position_id.name)
+        )
+
+        if len(currency_names) > 1:
+            error_msg += "{} {}: {}\n".format(
+                start_msg, _("currencies"), ", ".join([str(x) for x in currency_names])
+            )
+        if len(picking_type_names) > 1:
+            error_msg += "{} {}: {}\n".format(
+                start_msg,
+                _("picking types"),
+                ", ".join([str(x) for x in picking_type_names]),
+            )
+        if len(incoterm_names) > 1:
+            error_msg += "{} {}: {}\n".format(
+                start_msg, _("incoterms"), ", ".join([str(x) for x in incoterm_names])
+            )
+        if len(payment_term_names) > 1:
+            error_msg += "{} {}: {}\n".format(
+                start_msg,
+                _("payment terms"),
+                ", ".join([str(x) for x in payment_term_names]),
+            )
+        if len(fiscal_position_names) > 1:
+            error_msg += "{} {}: {}\n".format(
+                start_msg,
+                _("fiscal positions"),
+                ", ".join([str(x) for x in fiscal_position_names]),
+            )
```

Comment:
This `start_msg` / `error_msg` hack might work on english, but some translations may require a different gramatical structure. So this string concatenation won't work for them. Plus, it'll be a nightmare to translate as separated strings.

I recommend the not lazy approach and translate the full string for each case.
Also, compare records .. not record names.

I'd do something like this:

```py
error_messages = []

currencies = purchase_orders.currency_id
if len(currencies) > 1:
    error_messages.append(
        _(
            "You can't merge purchase orders with different currencies: %s",
            ", ".join(currencies.mapped("name"))
        )
    )

picking_types = purchase_orders.picking_type_id
if len(picking_types) > 1:
    error_messages.append(
        _(
            "You can't merge purchase orders with different picking types: %s",
            ", ".join(picking_types.mapped("name"))
        )
    )

# ... all the other checks ...

if error_messages:
    raise UserError("\n".join(error_messages))
```